src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodWithGCInfo.cs

```diff
@@ -42,6 +43,11 @@ public void SetCode(ObjectData data)
             _methodCode = data;
         }
 
+        public bool AlreadyStartedCompilation()
```

Comment:
Nit - the name of the method sounds like it's a plain property check while in reality it modifies the instance state machine; please consider indicating that somehow in the name, e.g. "CheckAndSetCompilationStarted".